### PR TITLE
chore(web): refactor asset and func id handling

### PR DESCRIPTION
**Pull request by paulocsanz**

### CODE REVIEW

File: app/web/src/store/asset.store.ts [hunk 1]
```diff
@@ -155,24 +135,26 @@ export const useAssetStore = () => {
           this.openAssetFuncIds[assetId] = funcs;
         },
 
-        closeFunc(assetId: AssetId, funcId: FuncId) {
+        CLOSE_FUNC(assetId: AssetId, funcId: FuncId) {
```
Comment (theoephraim) [inline app/web/src/store/asset.store.ts:138]: the convention here (which has already been broken in a few places) is that the ALL_UPPER functions are api action (ie they make an api request, and so will have a request status available).

File: app/web/src/store/asset.store.ts [hunk 2]
```diff
@@ -91,23 +90,22 @@ export const useAssetStore = () => {
...
-        selectedAssetId: null as AssetId | null,
         openAssetFuncIds: {} as { [key: AssetId]: FuncId[] },
       }),
       getters: {
         assets: (state) => _.values(state.assetsById),
         selectedAsset(): Asset | undefined {
           return this.assetsById[this.selectedAssetId || 0];
         },
-        urlSelectedAssetId(): AssetId | undefined {
+        selectedAssetId(): AssetId | undefined {
```
Comment (theoephraim) [inline app/web/src/store/asset.store.ts:100]: In the other stores, I've been careful to leave the name as `urlSelectedThingId` to make it clear that this is the one selected via the URL only. Then I usually have another `selectedThingId` which is the id of the thing once we have verified the id is good and loaded the thing.

This helps deal with the situation where there is an ID in the url but it isn't good (either totally bogus, or thing has been deleted, no access, etc...)

File: app/web/src/store/asset.store.ts (hunk 1 above)
Comment (theoephraim) [reply app/web/src/store/asset.store.ts:138]: yes plz. lets try to keep the UPPER_CASE actions only things that call the api

**Comment by paulocsanz** [comment]
bors merge

**Comment by paulocsanz** [comment]
bors merge

**Pull request merged**